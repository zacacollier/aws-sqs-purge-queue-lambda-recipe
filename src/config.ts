class Config {
  queueUrl: string;
  region: string;
  constructor() {
    this.queueUrl = process.env.QUEUE_URL as string;
    this.region = process.env.AWS_REGION as string;
  }
}

let config: Config;

export const getConfig = (): Config => {
  if (!config) {
    config = new Config();
  }
  return config;
};
