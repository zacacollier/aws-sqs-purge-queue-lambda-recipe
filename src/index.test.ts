import * as AWSLambda from 'aws-lambda';
import * as AWS from 'aws-sdk';
import { getConfig } from './config';
import { handler } from '.';

jest.mock('./config');
jest.mock('aws-sdk', () => {
  return {
    ...jest.requireActual('aws-sdk'),
    SQS: jest.fn(),
  };
});

const mocks = {
  SQSClient: AWS.SQS as unknown as jest.Mock,
  getConfig: getConfig as jest.Mock,
};

const event = {} as AWSLambda.APIGatewayEvent;
const context = {} as AWSLambda.Context;
const callback = new Function() as AWSLambda.Callback;
const queueUrl = 'QUEUE_URL';
const awsRegion = 'AWS_REGION';

describe('handler', () => {
  const awsSdkPromiseMock = jest.fn();
  const sqsPurgeQueueMock = jest
    .fn()
    .mockReturnValue({ promise: awsSdkPromiseMock });

  beforeAll(() => {
    mocks.getConfig.mockReturnValue({
      queueUrl,
      awsRegion,
    });
    mocks.SQSClient.mockImplementation(() => {
      return {
        purgeQueue: sqsPurgeQueueMock,
      };
    });
  });

  test('happy path', async () => {
    await expect(handler(event, context, callback)).resolves.toMatchObject({
      success: true,
    });
    expect(getConfig).toHaveBeenCalled();
    expect(sqsPurgeQueueMock.mock.calls).toMatchSnapshot();
  });

  test('error handled', async () => {
    awsSdkPromiseMock.mockRejectedValueOnce(
      new Error('Service unavailable') as never,
    );
    await expect(handler(event, context, callback)).resolves.toMatchObject({
      success: false,
    });
    expect(getConfig).toHaveBeenCalled();
    expect(sqsPurgeQueueMock.mock.calls).toMatchSnapshot();
  });
});
