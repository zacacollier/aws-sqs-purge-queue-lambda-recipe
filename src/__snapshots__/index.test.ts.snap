// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler error handled 1`] = `
[
  [
    {
      "QueueUrl": "QUEUE_URL",
    },
  ],
  [
    {
      "QueueUrl": "QUEUE_URL",
    },
  ],
]
`;

exports[`handler happy path 1`] = `
[
  [
    {
      "QueueUrl": "QUEUE_URL",
    },
  ],
]
`;
